### PR TITLE
add `runt environments`

### PR DESCRIPTION
**Pull request by cvaske**
still need to fix:
- prints out the displayname and language, rather than the dirname like jupyter does
- the kernelspecs() primary interface should return the kernelspec directory in additon to the json structure

### CODE REVIEW

**Comment by cvaske** [comment]
It feels like `list_kernelspec_names_at` at lines 61:75 in runtimelib/src/jupyter/kernelspec.rs could be more idiomatic, but I'm not sure how yet.

All comments welcome!

File: runtimelib/tests/kernels/ir/kernel.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
+  "argv": [
+    "/Library/Frameworks/R.framework/Resources/bin/R",
```
Comment (rgbkrk) [inline runtimelib/tests/kernels/ir/kernel.json:3]: With these checked in files you _might_ be able to allow these to be just a flat `R` as the first argument. Same for the execvr below.

**Comment by rgbkrk** [comment]
One area that I'd like to differ with Jupyter is making the public name for this feature be environments. For Jupyter we'll use kernelspecs as the spec is there, while leaving ourselves flexible for adapting to multiple python environments (whether pip or conda based) and even remote ones.

File: runtimelib/src/jupyter/kernelspec.rs [hunk 2]
```diff
@@ -0,0 +1,164 @@
+use std::fs::{read_dir, File};
```
Comment (rgbkrk) [inline runtimelib/src/jupyter/kernelspec.rs:1]: Switch over to using `tokio::fs` as Tokio is the async runtime we're using.

**Pull request merged**